ntp-proto/src/algorithm/kalman/select.rs

```diff
@@ -38,30 +43,44 @@ pub(super) fn select<Index: Copy>(
 
     bounds.sort_by(|a, b| a.0.total_cmp(&b.0));
 
-    let mut max: usize = 0;
-    let mut maxt: f64 = 0.0;
+    let mut maxlow: usize = 0;
+    let mut maxhigh: usize = 0;
+    let mut maxtlow: f64 = 0.0;
+    let mut maxthigh: f64 = 0.0;
     let mut cur: usize = 0;
 
     for (time, boundtype) in bounds.iter() {
         match boundtype {
-            BoundType::Start => cur += 1,
-            BoundType::End => cur -= 1,
-        }
-        if cur > max {
-            max = cur;
-            maxt = *time;
+            BoundType::Start => {
+                cur += 1;
+                if cur > maxlow {
+                    maxlow = cur;
+                    maxtlow = *time;
+                }
+            }
+            BoundType::End => {
+                if cur > maxhigh {
+                    maxhigh = cur;
+                    maxthigh = *time;
+                }
+                cur -= 1;
+            }
         }
     }
 
+    // Catch programming errors. If this ever fails there is high risk of missteering, better fail hard in that case
+    assert_eq!(maxlow, maxhigh);
+    let max = maxlow;
+
     if max >= synchronization_config.minimum_agreeing_sources && max * 4 > bounds.len() {
```

Comment:
I think we should probably update the documentation of minimum agreeing sources to indicate that it refers to non-periodic sources only?